File: spring-boot-swagger/src/main/java/com/example/sampleswagger/rest/api/SampleApiController.java
package com.example.sampleswagger.rest.api;

import com.example.sampleswagger.security.LoginUser;
import io.swagger.annotations.Api;
import io.swagger.annotations.ApiOperation;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.http.MediaType;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.*;

import javax.annotation.Resource;
import javax.validation.Valid;
import java.util.Objects;

@Api("サンプルAPI")
@RestController
@RequestMapping(path = "/api/sample", produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
public class SampleApiController {
    private final Logger logger = LoggerFactory.getLogger(this.getClass());
    @Resource
    private AsyncSupport asyncSupport;

    @ApiOperation("sample get async api")
    @GetMapping("public/async")
    public SampleApiResponse getAsync() {
        this.asyncSupport.sleep();
        SampleApiResponse response = new SampleApiResponse();
        this.logger.info("return response {}", response);
        return response;
    }

    @ApiOperation("sample get public api")
    @GetMapping("public")
    public SampleApiResponse getPublic(@ModelAttribute @Valid SampleApiRequest request) {
        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
        String detail = "has authentication: " + Objects.nonNull(authentication);
        return new SampleApiResponse(30, 35, "get", "sample public api", request.getKeyWord(), detail);
    }

    @ApiOperation("sample post public api")
    @PostMapping("public")
    public SampleApiResponse postPublic(@RequestBody @Valid SampleApiRequest request) {
        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
        String detail = "has authentication: " + Objects.nonNull(authentication);
        return new SampleApiResponse(30, 35, "post", "sample public api", request.getKeyWord(), detail);
    }

    @ApiOperation("sample get authenticated api")
    @GetMapping("auth")
    public SampleApiResponse getAuthenticated(
            @RequestParam(name = "page", defaultValue = "1", required = false) int page,
            @RequestParam(name = "size", defaultValue = "1", required = false) int size,
            @RequestParam(name = "keyword", defaultValue = "", required = false) String keyword) {
        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
        LoginUser loginUser = (LoginUser) authentication.getPrincipal();
        return new SampleApiResponse(30, 35, "get", "sample authenticated api", keyword, loginUser.toString());
    }

    @ApiOperation("sample post authenticated api")
    @PostMapping("auth")
    public SampleApiResponse postAuthenticated(@RequestBody @Valid SampleApiRequest request) {
        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
        LoginUser loginUser = (LoginUser) authentication.getPrincipal();
        return new SampleApiResponse(30, 35, "post", "sample authenticated api", request.getKeyWord(), loginUser.toString());
    }


}

File: spring-boot-rest-api/src/main/kotlin/com/yamashiro0110/restapisample/SampleApiController.kt
package com.yamashiro0110.restapisample

import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.databind.PropertyNamingStrategy
import com.fasterxml.jackson.databind.annotation.JsonNaming
import org.hibernate.validator.constraints.NotBlank
import org.springframework.web.bind.annotation.*
import java.time.ZoneId
import java.time.ZonedDateTime
import java.time.format.DateTimeFormatter
import javax.validation.Valid

@RestController
@RequestMapping("/v1/sample")
class SampleApiController {
    @GetMapping
    fun get() = SampleApiResponse(msg = "hello kotlin rest api sample!!", path = "v1/sample", method = "get")

    @PostMapping
    fun post(@RequestBody @Valid request: SampleApiRequest) = SampleApiResponse(request)
}

data class SampleApiRequest(
        @get:NotBlank
        var msg: String = "",
        var path: String = "",
        var method: String = ""
)

@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy::class)
data class SampleApiResponse(
        val msg: String,
        val path: String,
        val method: String) {

    constructor(request: SampleApiRequest) : this(
            msg = request.msg,
            path = request.path,
            method = request.method
    )

    fun zoneJst() = ZoneId.of(ZoneId.SHORT_IDS["JST"])

    fun format() = DateTimeFormatter.ISO_DATE_TIME

    @JsonProperty
    fun isoDateTime() = ZonedDateTime.now(zoneJst()).format(format())

    @JsonProperty
    fun dateTimeOfSystemTimeZone() = ZonedDateTime.now(ZoneId.systemDefault()).format(format())
}
